docs/source/features/memory_optimizations.rst

```diff
@@ -5,6 +5,80 @@ Parallelism
 -----------
 Refer to :doc:`Parallelism <./parallelisms>`.
 
+
+Mixture of Experts
+------------------
+
+Overview
+^^^^^^^^
+
+NeMo supports Mixture of Experts (MoE)in the transformer layer for NLP models.
+
+MoE is a machine learning technique where multiple specialized models (experts,
+usually multi-layer perceptrons) are combined to solve a complex task. Each expert
+focuses on a specific subtask or domain, while a gating network dynamically activates
+the most appropriate expert based on the current input.
+
+
+To use MoE  in the NeMo Framework, adjust the ``num_moe_experts`` parameter in the model configuration:
+
+1. Set ``num_moe_experts`` to `8` to leverage 8 experts in the MoE module.
+
+   .. code-block:: yaml
+
+       num_moe_experts: 8  # Set MoE to use 8 experts
+
+2. Set ``moe_router_topk`` to the number of experts you want activated, for eaxmple if you want to process each input with two experts:
+
+   .. code-block:: yaml
+
+       moe_router_topk: 2  # Processes each token using 2 experts.
+
+In addition NeMo provides options to configure MoE-specific loss function, for example to balance token
+distribution across experts:
+
+1. Set ``moe_router_load_balancing_type`` to specify the load balancing method:
+
+   .. code-block:: yaml
+
+      moe_router_load_balancing_type: aux_loss  # to use the auxilary loss, other options include "sinkhorn".
+
+2. Set ``moe_aux_loss_coeff`` to specify the weight of the auxilary loss, values in the 1e-2 range are a good start:
+
+   .. code-block:: yaml
+
+      moe_aux_loss_coeff: 1e-2  # set the aux-loss weight to 1e-2
+
+3. Set ``moe_z_loss_coeff`` to specify the weight of the z-loss, a starting value of 1e-3 is recommended:
```

Comment:
Fix punctuation and add sentence.

3. Set ``moe_z_loss_coeff`` to specify the weight of the z-loss.  A starting value of 1e-3 is recommended, as follows: